invenio_files_rest/views.py

```diff
@@ -594,6 +594,17 @@ def delete_object(self, bucket, obj, version_id):
             db.session.commit()
             if obj.file_id:
                 remove_file_data.delay(str(obj.file_id))
+            # Set newest object as head
+            if obj.is_head:
+                try:
+                    obj_to_restore = \
+                        ObjectVersion.get_versions(obj.bucket,
+                                                   obj.key,
+                                                   desc=True).first()
+                    obj_to_restore.is_head = True
+                    db.session.commit()
+                except AttributeError:
```

Comment:
What happens if you have only one version that you delete, and thus there's no version to restore. I think the ``.first()`` is going to throw a different error or return none? 

I think it would be more readable to do something like:
```python
obj_to_restore = ObjectVersion.get_versions(...)
if obj_to_restore:
     obj_to_restore.is_head = True
```

Reply:
> first()
> Return the first result of this Query or None if the result doesn’t contain any row.

Should be fine now.